src/libraries/Directory.Build.targets

```
@@ -244,7 +244,7 @@
 
   <!-- Adds ObsoleteAttribute to projects that need to apply downlevel Obsoletions with DiagnosticId and UrlFormat -->
   <Choose>
-    <When Condition="'$(IncludeInternalObsoleteAttribute)' == 'true' and ($(TargetFramework.StartsWith('netstandard')) or $(TargetFramework.StartsWith('netcoreapp')) or '$(TargetFrameworkIdentifier)' == '.NETFramework')">
+    <When Condition="'$(IncludeInternalObsoleteAttribute)' == 'true' and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
```

I believe this should make the condition even easier:

```suggestion
    <When Condition="'$(IncludeInternalObsoleteAttribute)' == 'true' and !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0'))">
```

---

For a better understanding of this function see my comment here: https://github.com/dotnet/msbuild/issues/5171#issuecomment-717233991

---

Woah; nice! Thanks, @ViktorHofer; I'll update this occurrence as well as one that exists just above that's the same, and I'll take a pass through looking for other places this can be used.

---

I didn't find any other occurrences of similar checks throughout our props/targets/*proj* files. Changed the two in this one.